### ISSUE TITLE
Alignment fails with "Aborted (core dumped)"

### ISSUE THREAD

**Issue by danoneata** [pykaldi/pykaldi#123]
Hello!

First of all, I want to thank you for the effort you put into this project! I'm sure that it will benefit many of us.

I was playing with the alignment functionality, but unfortunately I'm running into some C++ errors. Below is the code that I'm using – it is very similar to the one provided in the [examples](https://github.com/pykaldi/pykaldi/blob/master/examples/scripts/alignment/gmm-aligner.py), except that I have also included per-speaker normalization on the MFCCs (`MFCC_TYPE = 2`):

```python
import os
import pdb
import sys
import time

from kaldi.alignment import GmmAligner
from kaldi.util.table import SequentialMatrixReader

MFCC_TYPE = int(sys.argv[1])

BASE_PATH = "."
EXPS_PATH = os.path.join(BASE_PATH, "exp", "mono")
DATA_PATH = os.path.join(BASE_PATH, "data")
LANG_PATH = os.path.join(BASE_PATH, "data", "lang")

text_path = os.path.join(DATA_PATH, "train", "text")

cmvn_scp = os.path.join(DATA_PATH, "train", "cmvn.scp")
feats_scp = os.path.join(DATA_PATH, "train", "feats.scp")
mfcc_conf = os.path.join(BASE_PATH, "conf", "mfcc.conf")
utt2spk = os.path.join(DATA_PATH, "train", "utt2spk")
wav_scp = os.path.join(DATA_PATH, "train", "wav.scp")

paths = {
    "model_rxfilename": os.path.join(EXPS_PATH, "final.mdl"),
    "tree_rxfilename": os.path.join(EXPS_PATH, "tree"),
    "lexicon_rxfilename": os.path.join(LANG_PATH, "L.fst"),
    "symbols_filename": os.path.join(LANG_PATH, "words.txt"),
    "disambig_rxfilename": os.path.join(LANG_PATH, "phones", "disambig.int"),
}

aligner = GmmAligner.from_files(**paths)

if MFCC_TYPE == 1:
    mfcc_path = (
        f"ark:compute-mfcc-feats --config={mfcc_conf} scp:{wav_scp} ark:-"
        " | apply-cmvn-sliding --cmn-window=10000 --center=true ark:- ark:-"
        " | add-deltas ark:- ark:- |"
    )
elif MFCC_TYPE == 2:
    mfcc_path = (
        f"ark,s,cs:apply-cmvn $cmvn_opts --utt2spk=ark:{utt2spk} scp:{cmvn_scp} scp:{feats_scp} ark:-"
        " | add-deltas $delta_opts ark:- ark:- |"
    )
else:
    assert False

with SequentialMatrixReader(mfcc_path) as f, open(text_path, "r") as t:
    for (feat_key, feat), line in zip(f, t):
        text_key, text = line.strip().split(None, 1)
        assert feat_key == text_key
        out = aligner.align(feat, text)
        print(feat_key, out["alignment"], flush=True)
```

For these tests, I'm using the TIMIT dataset and the default Kaldi recipe [`egs/timit/s5/run.sh`](https://github.com/pykaldi/kaldi/blob/master/egs/timit/s5/run.sh).

I have observed the following behaviour:

1. Running `python main.py 1 2> /tmp/error1.log` fails with `Error in python: munmap_chunk(): invalid pointer: 0x0000000002adda50` and produces alignments for the first five utterances. [Here](https://pastebin.com/pzRH1Nt6) is the full error log.
2. Running `python main.py 2 2> /tmp/error2.log` fails with `Error in python: free(): invalid next size (fast): 0x0000000002fe8400` and produces no alignments. [Here](https://pastebin.com/bwPAKiuH) is the full error log.
3. Running `python main.py 2 > /tmp/out2.log 2> /tmp/error2.log` fails with the same the error as above, but produces alignments for 18 utterances.

**Comment by vrmpx** [comment]
Hello, are you using a docker container with a limited amount of memory? Are you consistently getting the same amount of utterances processed in each case?

**Comment by danoneata** [comment]
No, I was not using the Docker container, but the code compiled from source. And I think here lies my problem. I have just checked the conda installation and the above snippet works fine.

I remember that I've used g++-5 to compile Kaldi (because of the CUDA 8.0 requirements), but I don't recall using it for CLIF. So I recompiled CLIF with the same version of C++ `CXX=g++-5 ./install_clif.sh` and it seems to work well now.

Thanks for pointing me in the right direction!
